packages/kit/src/runtime/client/client.js

```diff
@@ -790,17 +785,29 @@ export function create_client({ target, base, trailing_slash }) {
 			if (loaders[i]) {
 				try {
 					branch.push(await branch_promises[i]);
-				} catch (e) {
-					const error = normalize_error(e);
-
-					if (error instanceof Redirect) {
+				} catch (err) {
+					if (err instanceof Redirect) {
 						return {
 							type: 'redirect',
-							location: error.location
+							location: err.location
 						};
 					}
 
-					const status = e instanceof HttpError ? e.status : 500;
+					let status = 500;
+					/** @type {App.PageError} */
+					let error;
+
+					if (/** @type {any} */ (err)?.type === 'error') {
```

Comment:
i think this should work, actually? bit cheeky but does the job with a minimum of fuss

```suggestion
					if (server_data_nodes.includes(err)) {
```

Reply:
i made this change, seems to work

Reply:
Ooh yeah this is smart 🤓